### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by OliverHerms**
Added some additional documentation to the top of the files. 

Added A label class, as well as a very basic GUI to start the test with a mouse click as an alternative. 

Some general rework from previous request´s comments.

### CODE REVIEW

File: PAT_bench.py [hunk 1]
```diff
@@ -1,65 +1,129 @@
...
+#    def opd3033_set_LowV(self):
+#        self.ODP3033.set_channel( 2, 4, 0.25, 0.3) ##Channel, Voltage, Current, Current-Limit
+#        if CONSOLE:    
+#            print('-----')
+#            print('low voltage set')
+#            print('-----') 
+#
+#    def opd3033_get_current(self, channel):
+#        measure = self.ODP3033.measure_current(channel)
+#        return measure
```
Comment (paul-cowboy) [inline PAT_bench.py:118]: Ideally you should keep this code around and select the right instrument depending on arguments to your program (e.g. passed through docopt). It should not require editing the code to switch between different devices

**Review by paul-cowboy — COMMENTED** [review]
Not done reviewing

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,6 @@
 *.log
 *.pyc
+*.txt
```
Comment (LaurentErnes) [inline .gitignore:3]: Don't you want test files ? Maybe in some sub-directory ?

File: PAT_bench.py [hunk 3]
```diff
@@ -1,65 +1,129 @@
...
+        self.DL3021.set_load('off')
+
+    def load_get_voltage(self):
+        voltage = self.DL3021.get_voltage()
+        return voltage
+
+    def load_get_power(self):
+        power = self.DL3021.get_power()
+        return power
+
```
Comment (LaurentErnes) [inline PAT_bench.py:56]: Maybe you should create a class `Load` that takes the `DL3021` and provides `turnOn()`, `turnOff()`, `voltage()` and `power()`

File: label.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        if self.print_both:
+            rem =  subprocess.Popen(['sudo','lp','-d','Zebra_ZD410','print_removable.txt'], stdout=subprocess.PIPE)
+            rem.stdout.read()
+            per =  subprocess.Popen(['sudo','lp','-d','Zebra_TLP_2824_Plus','print_permanent.txt'], stdout=subprocess.PIPE)
+            per.stdout.read()
+        else:
+            per =  subprocess.Popen(['sudo','lp','-d','Zebra_TLP_2824_Plus','print_permanent.txt'], stdout=subprocess.PIPE)
+            per.stdout.read()
+
+
```
Comment (LaurentErnes) [inline label.py:47]: Good.
TIPS: if you want to unit test a class, you can use the following block that runs only if the script is run as main:
```
if __name__ == "__main__":
  # run some unit test here
  label = Label()
  ...
```

Comment (LaurentErnes) [reply label.py:47]: then you just need to invoke `python label.py`

File: main.py [hunk 5]
```diff
@@ -52,124 +72,176 @@ def setup_logger(debug):
...
+            self.interface_name)
+        try:
+            if os.system(cmd) != 0: # This will run the command and check connection
+                raise Exception()
+        except:
+            raise # Not Connected
+        else:
+            return True # Connected
+########################################################
+
```
Comment (LaurentErnes) [inline main.py:162]: This part should be moved to some dedicated files.
I don't like `main.py` to be so cluttered.

**Review by LaurentErnes — COMMENTED** [review]
I think the code is missing some level of abstraction, and a bit of cleanup.
Apart from that, it should be functional; I have no way to test it.

File: label.py (hunk 4 above)
Comment (OliverHerms) [reply label.py:47]: Ah yes, I´ve seen that before, makes sense! That is a good tip!

File: main.py (hunk 5 above)
Comment (OliverHerms) [reply main.py:162]: True, not planning on keeping this here. 

BTW: Do you have experience with checking internet connection with pyhton on Linux?

File: .gitignore (hunk 2 above)
Comment (OliverHerms) [reply .gitignore:3]: In this case, nothing stored in .txt files is needed publicly on GitHub or as part of the Code.

**Pull request closed without merging**